gtsam/geometry/tests/testPose3.cpp

```diff
@@ -878,6 +878,105 @@ TEST(Pose3 , ChartDerivatives) {
   }
 }
 
+//******************************************************************************
+#include "testPoseAdjointMap.h"
+
+TEST(Pose3, TransformCovariance6MapTo2d) {
+  // Create 3d scenarios that map to 2d configurations and compare with Pose2 results.
+  using namespace test_pose_adjoint_map;
+
+  Vector3 s2{0.1, 0.3, 0.7};
+  Pose2 p2{1.1, 1.5, 31. * degree};
+  auto cov2 = FullCovarianceFromSigmas<Pose2>(s2);
+  auto transformed2 = Pose2::TransformCovariance{p2}(cov2);
+
+  auto match_cov3_to_cov2 = [&](int spatial_axis0, int spatial_axis1, int r_axis,
+                                const Pose2::Jacobian &cov2, const Pose3::Jacobian &cov3) -> void
+  {
+    EXPECT(assert_equal(
+      Vector3{cov2.diagonal()},
+      Vector3{cov3(spatial_axis0, spatial_axis0), cov3(spatial_axis1, spatial_axis1), cov3(r_axis, r_axis)}));
+    EXPECT(assert_equal(
+      Vector3{cov2(1, 0), cov2(2, 0), cov2(2, 1)},
+      Vector3{cov3(spatial_axis1, spatial_axis0), cov3(r_axis, spatial_axis0), cov3(r_axis, spatial_axis1)}));
+  };
+
+  // rotate around x axis
+  {
+    auto cov3 = FullCovarianceFromSigmas<Pose3>((Vector6{} << s2(2), 0., 0., 0., s2(0), s2(1)).finished());
+    auto transformed3 = Pose3::TransformCovariance{{Rot3::RzRyRx(p2.theta(), 0., 0.), {0., p2.x(), p2.y()}}}(cov3);
```

Comment:
Here the argument to FullCovarianceFromSigmas<Pose3>() needs to be constructed with a Vector6 comma initializer. Yet FullCovarianceFromSigmas<Pose2>() can use 3 comma separated doubles. It would be nice to use 6 comma separated doubles in this case as well.

Reply:
Not super-critical.